core/src/main/java/io/fabric8/maven/core/util/MavenUtil.java

```diff
@@ -204,6 +222,78 @@ public static Properties getPropertiesFile(MavenProject project, String properti
+     */
+    public static Properties getPropertiesFromYamlFile(MavenProject project, String yamlFileName) {
+        URLClassLoader compileClassLoader = getCompileClassLoader(project);
+        URL resource = compileClassLoader.findResource(yamlFileName);
+        return getPropertiesFromYamlResource(resource);
+    }
+
+    /**
+     * Returns a {@code Properties} representation of the given Yaml resource or an empty properties object if the resource is null
+     */
+    protected static Properties getPropertiesFromYamlResource(URL resource) {
+        if (resource != null) {
+            try (InputStream yamlStream = resource.openStream()) {
+                Yaml yaml = new Yaml();
+                @SuppressWarnings("unchecked")
+                SortedMap<String, Object> source = yaml.loadAs(yamlStream, SortedMap.class);
+                Properties properties = new Properties();
+                properties.putAll(getFlattenedMap(source));
+                return properties;
+            } catch (Exception e) {
+                LOG.error("Failed to load yaml properties from: " + resource + ". " + e, e);
+            }
+        }
+        return new Properties();
+    }
+
+    /**
+     * Build a flattened representation of the Yaml tree. The conversion is compliant with the spring-boot rules.
+     */
+    private static Map<String, Object> getFlattenedMap(Map<String, Object> source) {
+        Map<String, Object> result = new LinkedHashMap<>();
+        buildFlattenedMap(result, source, null);
+        return result;
+    }
+
+    @SuppressWarnings("unchecked")
+    private static void buildFlattenedMap(Map<String, Object> result, Map<String, Object> source, String path) {
+        for (Map.Entry<String, Object> entry : source.entrySet()) {
+            String key = entry.getKey();
+            if (path !=null && path.trim().length()>0) {
+                if (key.startsWith("[")) {
+                    key = path + key;
+                }
+                else {
+                    key = path + "." + key;
+                }
+            }
+            Object value = entry.getValue();
+            if (value instanceof String) {
+                result.put(key, value);
+            }
+            else if (value instanceof Map) {
+
+                Map<String, Object> map = (Map<String, Object>) value;
+                buildFlattenedMap(result, map, key);
+            }
+            else if (value instanceof Collection) {
+                Collection<Object> collection = (Collection<Object>) value;
+                int count = 0;
+                for (Object object : collection) {
+                    buildFlattenedMap(result,
+                            Collections.singletonMap("[" + (count++) + "]", object), key);
+                }
+            }
+            else {
+                result.put(key, (value != null ? value : ""));
```

Comment:
Question is whether we would add an null value as empty string or simply skip it, if null. have no strong opinion on that, though.